compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -1780,14 +1780,21 @@ trait Applications extends Compatibility {
       def winsType2 = isAsSpecific(alt2, tp2, alt1, tp1)
 
       overload.println(i"compare($alt1, $alt2)? $tp1 $tp2 $ownerScore $winsType1 $winsType2")
-      if (ownerScore == 1)
-        if (winsType1 || !winsType2) 1 else 0
-      else if (ownerScore == -1)
-        if (winsType2 || !winsType1) -1 else 0
-      else if (winsType1)
-        if (winsType2) 0 else 1
+      if winsType1 && winsType2
+          && alt1.widenExpr =:= alt2.widenExpr
+          && alt1.widenExpr.isStable
```

Comment:
Check is `isStable` first, for performance?

Reply:
Yes, good idea.